core/src/main/scala/kafka/coordinator/group/CoordinatorLoaderImpl.scala

```diff
@@ -135,7 +136,22 @@ class CoordinatorLoaderImpl[T](
 
             memoryRecords.batches.forEach { batch =>
               if (batch.isControlBatch) {
-                throw new IllegalStateException("Control batches are not supported yet.")
+                batch.asScala.foreach { record =>
```

Comment:
This updates the loader to handle control records.

Reply:
Is it always the case that we need to complete the transaction on loading?
Is this because loading rebuilds memory from the log so we play through the whole transaction again?
Maybe I'm just getting confused by the "completeTransaction" method names as well.

Reply:
@jolshan If we have a control record for committing or aborting the transaction, my understanding is that we must apply it. This means that all the pending transactional offsets seen so far must be committed. Do you have a suggestion for a better name? How about `replayEndTransactionMarker`?

@jeffkbkim What is it confusing? From a state machine perspective, the transaction is completed.

Reply:
I don't see a reason for having both "apply to state machine" and "write marker to log" to be the same name. in `CoordinatorTransactionEndMarkerEvent` we use both methods together. It's hard to follow at least for me

Reply:
`replayEndTransactionMarker` seems like a good compromise.